cli/azd/pkg/environment/environment.go

```diff
@@ -134,65 +121,6 @@ func CleanName(name string) string {
 	return result.String()
 }
 
-// FromRoot loads an environment located in a directory. On error,
-// an valid empty environment file, configured to persist its contents
-// to this directory, is returned.
-func FromRoot(root string) (*Environment, error) {
-	if _, err := os.Stat(root); err != nil {
-		return EmptyWithRoot(root), err
-	}
-
-	env := &Environment{
-		Root: root,
-	}
-
-	if err := env.Reload(); err != nil {
-		return EmptyWithRoot(root), err
-	}
-
-	return env, nil
-}
-
-func GetEnvironment(azdContext *azdcontext.AzdContext, name string) (*Environment, error) {
-	return FromRoot(azdContext.EnvironmentRoot(name))
-}
-
-// EmptyWithRoot returns an empty environment, which will be persisted
-// to a given directory when saved.
-func EmptyWithRoot(root string) *Environment {
-	return &Environment{
-		Root:        root,
-		dotenv:      make(map[string]string),
-		deletedKeys: make(map[string]struct{}),
-		Config:      config.NewEmptyConfig(),
-	}
-}
-
-// Ephemeral returns returns an empty ephemeral environment (i.e. not backed by a file) with a set
-func Ephemeral() *Environment {
-	return &Environment{
-		dotenv:      make(map[string]string),
-		deletedKeys: make(map[string]struct{}),
-		Config:      config.NewEmptyConfig(),
-	}
-}
-
-// EphemeralWithValues returns an ephemeral environment (i.e. not backed by a file) with a set
-// of values. Useful for testing. The name parameter is added to the environment with the
-// AZURE_ENV_NAME key, replacing an existing value in the provided values map. A nil values is
-// treated the same way as an empty map.
-func EphemeralWithValues(name string, values map[string]string) *Environment {
-	env := Ephemeral()
-
-	if values != nil {
-		env.dotenv = values
-	}
-
-	env.dotenv[EnvNameEnvVarName] = name
-
-	return env
-}
```

Comment:
`environment` objects are always ephemeral now and the loading/saving aspects have been moved to the environment `DataStore` implementations.